### PR TITLE
Modify test_positive_import_all_roles to cover non-admin scenario

### PR DESCRIPTION
**Pull request by Gauravtalreja1**
### Problem Statement
Currently, test_positive_import_all_roles covers only admin scenario to import all roles via UI, and as part of component audit its suggest to modify this test to cover non-admin user scenario

### Solution
Parametrizing the existing test to cover both admin and non-admin user scenarios with few more changes

### CODE REVIEW

**Comment by Gauravtalreja1** [comment]
trigger: test-robottelo
pytest: tests/foreman/destructive/test_ansible.py::test_positive_import_all_roles

**Comment by Satellite-QE** [comment]
**PRT Result**
```
Build Number: 7108
Build Status: UNSTABLE
PRT Comment: pytest tests/foreman/destructive/test_ansible.py::test_positive_import_all_roles --external-logging
Test Result : ================= 176 warnings, 2 errors in 2007.16s (0:33:27) =================
```

**Comment by Gauravtalreja1** [comment]
trigger: test-robottelo
pytest: tests/foreman/destructive/test_ansible.py::test_positive_import_all_roles

**Comment by Satellite-QE** [comment]
**PRT Result**
```
Build Number: 7115
Build Status: UNSTABLE
PRT Comment: pytest tests/foreman/destructive/test_ansible.py::test_positive_import_all_roles --external-logging
Test Result : ================= 173 warnings, 2 errors in 1979.93s (0:32:59) =================
```

**Comment by Gauravtalreja1** [comment]
trigger: test-robottelo
pytest: tests/foreman/destructive/test_ansible.py::test_positive_import_all_roles

**Comment by Satellite-QE** [comment]
**PRT Result**
```
Build Number: 7148
Build Status: UNSTABLE
PRT Comment: pytest tests/foreman/destructive/test_ansible.py::test_positive_import_all_roles --external-logging
Test Result : ================= 181 warnings, 2 errors in 2063.42s (0:34:23) =================
```

**Comment by Gauravtalreja1** [comment]
trigger: test-robottelo
pytest: tests/foreman/destructive/test_ansible.py::test_positive_import_all_roles

**Comment by Satellite-QE** [comment]
**PRT Result**
```
Build Number: 7150
Build Status: SUCCESS
PRT Comment: pytest tests/foreman/destructive/test_ansible.py::test_positive_import_all_roles --external-logging
Test Result : ================= 2 passed, 187 warnings in 2261.01s (0:37:41) =================
```

**Review by synkd — APPROVED** [review]
This looks good, so ACK.

Out of curiosity, though, what did you need to change to get PRT passing? I looked at one of the failures this morning and it appeared to be an issue with the `function_org` fixture.

**Comment by Gauravtalreja1** [comment]
Yes, the PRT failure earlier was due to https://github.com/SatelliteQE/nailgun/issues/1038, which was fixed by @tpapaioa  in https://github.com/SatelliteQE/nailgun/pull/1152, Thanks!

File: tests/foreman/destructive/test_ansible.py [hunk 1]
```diff
@@ -63,7 +66,23 @@ def test_positive_import_all_roles(target_sat):
...
+        username = gen_string('alphanumeric')
+        user = target_sat.api.User(
+            role=ansible_manager_role,
+            admin=True,
+            login=username,
+            password=password,
+            organization=[function_org],
+        ).create()
+        request.addfinalizer(user.delete)
+    username = user.login if auth_type == 'non-admin' else settings.server.admin_username
```
Comment (omkarkhatavkar) [inline tests/foreman/destructive/test_ansible.py:83]: above if statement can be merged in the previous if statement it self, non-blocking

**Review by shubhamsg199 — APPROVED** [review]

File: tests/foreman/destructive/test_ansible.py (hunk 1 above)
Comment (Gauravtalreja1) [reply tests/foreman/destructive/test_ansible.py:83]: @omkarkhatavkar Good point, updated!

**Comment by Gauravtalreja1** [comment]
trigger: test-robottelo
pytest: tests/foreman/destructive/test_ansible.py::test_positive_import_all_roles

**Comment by Satellite-QE** [comment]
**PRT Result**
```
Build Number: 7221
Build Status: SUCCESS
PRT Comment: pytest tests/foreman/destructive/test_ansible.py::test_positive_import_all_roles --external-logging
Test Result : ================= 2 passed, 194 warnings in 2204.28s (0:36:44) =================
```

**Pull request merged**